### PR TITLE
Feature/44 exclude segs

### PR DESCRIPTION
**Pull request by markhayden**
Added ability to override lytics api and css file locally without modifying main pathfora.js file. Added testing build and default values for api. 

Added ability to exclude segments as well as testing for those scenarios and finally an example file for how to config the excludes. 

Desired behavior is to prevent displaying widgets for any users that have explicitly been excluded. for example, if i am in the "new_user" segment but also in the "converted" segment, we might want to suppress a message saying "make your first purchase". This could also be done through segments but that adds much more complexity.

### CODE REVIEW

File: src/pathfora.js [hunk 1]
```diff
@@ -398,17 +399,29 @@
...
       }
+      if(!config.exclude){
+        throw new Error('Invalid scaffold. No exclude array.');
+      }
+
+      if (method === "target"){
+        subject = config.target;
+      }else if(method === "exclude"){
+        subject = config.exclude;
+      }else{
```
Comment (sarahyablok) [inline src/pathfora.js]: Should we also check for `inverse` method here?

File: test/pathforaSpec.js [hunk 2]
```diff
@@ -105,6 +105,59 @@ describe('Pathfora', function () {
     jasmine.Ajax.uninstall();
   });
 
+  it('bonk bnok bonok', function (done) {
```
Comment (sarahyablok) [inline test/pathforaSpec.js]: This test should have a real description.

File: src/pathfora.js (hunk 1 above)
Comment (sarahyablok) [inline src/pathfora.js]: We might also support the `common` method? :horse:

File: test/pathforaSpec.js (hunk 2 above)
Comment (markhayden) [inline test/pathforaSpec.js]: hahahahahahaahahaha nice. i was struggling to get things to work, apparently i forgot to fix that once i got it straightened out :)

File: src/pathfora.js (hunk 1 above)
Comment (markhayden) [inline src/pathfora.js]: yah i wasnt even sure what common was doing so didnt include it in the scaffold.

**Comment by sarahyablok** [comment]
:+1:

**Pull request merged**